plugins/modules/dcnm_vrf.py

```diff
@@ -429,10 +435,11 @@ def diff_for_attach_deploy(self, want_a, have_a):
                         else:
                             found = True
 
-                            if bool(have['isAttached']) is not bool(want['isAttached']):
-                                del want['isAttached']
-                                attach_list.append(want)
-                                continue
+                            if want.get('isAttached') is not None:
+                                if bool(have['isAttached']) is not bool(want['isAttached']):
```

Comment:
bool(True) is True
bool('False') is also True

Not sure what you are checking here?

Reply:
There was this traceback when we try to delete a vrf which is not attached to any switch as the key 'isAttached' was not there in the want dict. want.get checks if the key is there in the dict, it returns the value if key is present, if not it returns none. The intention here is to continue with our check if 'isAttached' is there in dict, else we skip the check.